nilearn/maskers/surface_masker.py

```diff
@@ -0,0 +1,486 @@
+"""Masker for surface objects."""
+
+from __future__ import annotations
+
+import warnings
+
+import numpy as np
+from joblib import Memory
+from sklearn.base import BaseEstimator, TransformerMixin
+
+from nilearn import signal
+from nilearn._utils import _constrained_layout_kwargs
+from nilearn._utils.cache_mixin import CacheMixin, cache
+from nilearn._utils.class_inspect import get_params
+from nilearn._utils.helpers import is_matplotlib_installed
+from nilearn.experimental.surface._surface_image import SurfaceImage
+from nilearn.maskers._utils import (
+    check_same_n_vertices,
+    compute_mean_surface_image,
+    get_min_max_surface_image,
+)
+
+
+class SurfaceMasker(TransformerMixin, CacheMixin, BaseEstimator):
+    """Extract data from a SurfaceImage.
+
+    Parameters
+    ----------
+    mask_img: SurfaceImage object or None, default=None
+
+    %(smoothing_fwhm)s
+        This parameter is not implemented yet.
+
+    %(standardize_maskers)s
+
+    %(standardize_confounds)s
+
+    %(detrend)s
+
+    high_variance_confounds : :obj:`bool`, default=False
+        If True, high variance confounds are computed on provided image with
+        :func:`nilearn.image.high_variance_confounds` and default parameters
+        and regressed out.
+
+    %(low_pass)s
+
+    %(high_pass)s
+
+    %(t_r)s
+
+    %(memory)s
+
+    %(memory_level1)s
+
+    %(verbose0)s
+
+    reports : :obj:`bool`, default=True
+        If set to True, data is saved in order to produce a report.
+
+    %(cmap)s
+        default=`inferno`
+        Only relevant for the report figures.
+
+    clean_kwargs : :obj:`dict` or None, default=None
+        Keyword arguments to be passed
+        to :func:`~nilearn.signal.clean`
+        called within the masker.
+
+    Attributes
+    ----------
+    output_dimension_: :obj:`int` or None
+        number of vertices included in mask
+
+    mask_img_: SurfaceImage or None
+    """
+
+    def __init__(
+        self,
+        mask_img=None,
```

Comment:
extra arguments that exist for NiftiMasker that could be introduced later:
 - mask_strategy: strategy used to compute the mask
 - mask_args: additional parameters passed to masking.compute_background_mask or masking.compute_epi_mask to fine-tune mask computation

may require masking.py to work on surface data

Reply:
Not so sure we want to have them here.  These functions are meant to extract automatically a brain mask from EPI data. I don't really see what it would bring on a brain surface mesh.

Reply:
hum... yeah I think this may not make much sense...